hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -162,15 +168,28 @@ public int getClientEndpointCount() {
 
     public void handleClientMessage(ClientMessage clientMessage, Connection connection) {
         int partitionId = clientMessage.getPartitionId();
-        final MessageTask messageTask = messageTaskFactory.create(clientMessage, connection);
+        MessageTask messageTask = messageTaskFactory.create(clientMessage, connection);
+        InternalOperationService operationService = nodeEngine.getOperationService();
         if (partitionId < 0) {
-            executor.execute(messageTask);
+            if (isUrgent(messageTask)) {
+                operationService.execute(new PriorityPartitionSpecificRunnable(messageTask));
```

Comment:
I don't get it. 
`partitionId < 0` (line 173) means it is NOT a partition specific task, doesn't it? Yet here you are creating `PriorityPartitionSpecificRunnable` 

`/me = confused`